File: sources/LiquidSense Release/me/aquavit/liquidsense/module/modules/render/KillESP.java
package me.aquavit.liquidsense.module.modules.render;

import me.aquavit.liquidsense.event.EventTarget;
import me.aquavit.liquidsense.module.modules.blatant.Aura;
import me.aquavit.liquidsense.LiquidSense;
import me.aquavit.liquidsense.event.events.PacketEvent;
import me.aquavit.liquidsense.event.events.Render3DEvent;
import me.aquavit.liquidsense.event.events.UpdateEvent;
import me.aquavit.liquidsense.event.events.WorldEvent;
import me.aquavit.liquidsense.module.Module;
import me.aquavit.liquidsense.module.ModuleCategory;
import me.aquavit.liquidsense.module.ModuleInfo;
import me.aquavit.liquidsense.utils.render.ColorUtils;
import me.aquavit.liquidsense.utils.render.RenderUtils;
import me.aquavit.liquidsense.value.BoolValue;
import me.aquavit.liquidsense.value.IntegerValue;
import me.aquavit.liquidsense.value.ListValue;
import net.minecraft.client.renderer.entity.RenderManager;
import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.entity.effect.EntityLightningBolt;
import net.minecraft.network.Packet;
import net.minecraft.network.play.client.C02PacketUseEntity;

import java.awt.*;
import java.util.HashMap;
import java.util.Map;
import java.util.Random;

@ModuleInfo(name = "KillESP", description = "KillESP", category = ModuleCategory.RENDER)
public class KillESP extends Module {
    private final ListValue modeValue = new ListValue("Mode", new String[]{"Crystal", "Box", "Head", "Mark"}, "Crystal");
    private final IntegerValue colorRedValue = new IntegerValue("R", 0, 0, 255);
    private final IntegerValue colorGreenValue = new IntegerValue("G", 160, 0, 255);
    private final IntegerValue colorBlueValue = new IntegerValue("B", 255, 0, 255);
    private final IntegerValue alphaValue = new IntegerValue("Alpha", 255, 0, 255);
    private final BoolValue killLightningBoltValue = new BoolValue("LightningBolt", true);
    private final BoolValue rainbow = new BoolValue("RainBow", false);
    private final BoolValue hurt = new BoolValue("HurtTime", true);

    private HashMap<EntityLivingBase, Long> targetList = new HashMap<EntityLivingBase, Long>();
    private Aura aura = (Aura) LiquidSense.moduleManager.getModule(Aura.class);
    Random random = new Random();

    @EventTarget
    public void onUpdate(UpdateEvent event) {
        if (killLightningBoltValue.get()) {
            for (Map.Entry<EntityLivingBase, Long> entry : targetList.entrySet()) {
                if ((entry.getKey()).isDead || (entry.getKey()).getHealth() == 0F) {
                    if(entry.getValue() > System.currentTimeMillis()){
                        EntityLightningBolt ent = new EntityLightningBolt(mc.theWorld, (entry.getKey()).posX, (entry.getKey()).posY, (entry.getKey()).posZ);
                        mc.theWorld.addEntityToWorld(-1, ent);
                        mc.thePlayer.playSound("random.explode", 0.5f, 0.5f + random.nextFloat() * 0.2f);
                    }
                    targetList.remove(entry.getKey());
                }
            }
        }
    }

    @EventTarget
    public void onPacket(PacketEvent event) {
        Packet<?> packet = event.getPacket();
        if (killLightningBoltValue.get()) {
            if (packet instanceof C02PacketUseEntity) {
                if (((C02PacketUseEntity) packet).getAction() == C02PacketUseEntity.Action.ATTACK) {
                    Entity entity = ((C02PacketUseEntity) packet).getEntityFromWorld(mc.theWorld);
                    if (entity instanceof EntityLivingBase) {
                        ((Map)targetList).put(entity, System.currentTimeMillis() + (long)3000);
                    }
                }
            }
        }
    }

    @EventTarget
    public void onWorld(WorldEvent event) {
        targetList.clear();
    }

    @EventTarget
    public void onRender3D(Render3DEvent event){
        Color color = rainbow.get() ? ColorUtils.rainbow() : new Color(colorRedValue.get(), colorGreenValue.get(), colorBlueValue.get(), alphaValue.get());
        RenderManager renderManager = mc.getRenderManager();
        EntityLivingBase entityLivingBase = aura.getTarget();
        if (entityLivingBase == null) return;
        double pX = (entityLivingBase.lastTickPosX + (entityLivingBase.posX - entityLivingBase.lastTickPosX) * mc.timer.renderPartialTicks
                - renderManager.renderPosX);
        double pY = (entityLivingBase.lastTickPosY + (entityLivingBase.posY - entityLivingBase.lastTickPosY) * mc.timer.renderPartialTicks
                - renderManager.renderPosY);
        double pZ = (entityLivingBase.lastTickPosZ + (entityLivingBase.posZ - entityLivingBase.lastTickPosZ) * mc.timer.renderPartialTicks
                - renderManager.renderPosZ);
        switch (modeValue.get().toLowerCase()){
            case "crystal":
                RenderUtils.drawCrystal(entityLivingBase, pX, pY + 2.25, pZ,(hurt.get() && entityLivingBase.hurtTime > 3) ? new Color(255, 50, 50, 75).getRGB() : color.getRGB(),(hurt.get() && entityLivingBase.hurtTime > 3) ? new Color(255, 50, 50, 75).getRGB() : color.getRGB());
                break;
            case "box":
                RenderUtils.drawEntityBoxESP(entityLivingBase,(hurt.get() && entityLivingBase.hurtTime > 3) ? new Color(255, 50, 50, 75) : color);
                break;
            case "head":
                RenderUtils.drawPlatformESP(entityLivingBase,(hurt.get() && entityLivingBase.hurtTime > 3) ? new Color(255, 50, 50, 75) : color);
                break;
            case "mark":
                RenderUtils.drawPlatform(entityLivingBase,  (hurt.get() && entityLivingBase.hurtTime > 3) ? new Color(255, 50, 50, 75) : color);
                break;
        }

    }
}

File: sources/LiquidSense Dev/me/AquaVit/liquidSense/modules/render/KillESP.kt
package me.AquaVit.liquidSense.modules.render

import net.ccbluex.liquidbounce.LiquidBounce
import net.ccbluex.liquidbounce.event.*
import net.ccbluex.liquidbounce.features.module.Module
import net.ccbluex.liquidbounce.features.module.ModuleCategory
import net.ccbluex.liquidbounce.features.module.ModuleInfo
import net.ccbluex.liquidbounce.features.module.modules.combat.Aura
import net.ccbluex.liquidbounce.utils.misc.MiscUtils.random
import net.ccbluex.liquidbounce.utils.render.ColorUtils.rainbow
import net.ccbluex.liquidbounce.utils.render.RenderUtils.*
import net.ccbluex.liquidbounce.value.BoolValue
import net.ccbluex.liquidbounce.value.IntegerValue
import net.ccbluex.liquidbounce.value.ListValue
import net.minecraft.entity.EntityLivingBase
import net.minecraft.entity.effect.EntityLightningBolt
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.network.play.client.C02PacketUseEntity
import java.awt.Color

@ModuleInfo(name = "KillESP", description = "KillESP", category = ModuleCategory.RENDER)
class KillESP : Module() {
    private val modeValue = ListValue("Mode", arrayOf("Crystal", "Box", "Head", "Mark"), "Crystal")

    private val colorRedValue = IntegerValue("R", 0, 0, 255)
    private val colorGreenValue = IntegerValue("G", 160, 0, 255)
    private val colorBlueValue = IntegerValue("B", 255, 0, 255)
    private val alphaValue = IntegerValue("Alpha", 255, 0, 255)
    private val killLightningBoltValue = BoolValue("LightningBolt", true)
    private val rainbow = BoolValue("RainBow", false)
    private val hurt = BoolValue("HurtTime", true)
    private val killAura = LiquidBounce.moduleManager.getModule(Aura::class.java) as Aura

    private var targetList: HashMap<EntityLivingBase, Long> = HashMap()

    @EventTarget
    fun onUpdate(event: UpdateEvent) {
        if (killLightningBoltValue.get()) {
            for (entry in targetList) {
                if (entry.key.isDead || entry.key.health == 0F) {
                    if(entry.value > System.currentTimeMillis()){
                        val ent = EntityLightningBolt(mc.theWorld, entry.key.posX, entry.key.posY, entry.key.posZ)
                        mc.theWorld.addEntityToWorld(-1, ent)
                        mc.thePlayer.playSound("random.explode", 0.5f, 0.5f + random.nextFloat() * 0.2f)
                    }
                    targetList.remove(entry.key)
                }
            }
        }
    }

    @EventTarget
    fun onPacket(event: PacketEvent) {
        if (killLightningBoltValue.get()) {
            if (event.packet is C02PacketUseEntity) {
                if (event.packet.action == C02PacketUseEntity.Action.ATTACK) {
                    val entity = event.packet.getEntityFromWorld(mc.theWorld)
                    if (entity is EntityLivingBase) {
                        targetList[entity] = System.currentTimeMillis() + 3000
                    }
                }
            }
        }
    }

    @EventTarget
    fun onWorldChange(event: WorldEvent) {
        targetList.clear()
    }

    @EventTarget
    fun onRender3D(event: Render3DEvent){
        var color = if (rainbow.get()) rainbow() else Color(colorRedValue.get(), colorGreenValue.get(), colorBlueValue.get(), alphaValue.get())
        killAura.target ?:return
        val RenderManager = mc.renderManager
        val entityLivingBase: EntityLivingBase = killAura.target as EntityLivingBase
        val pX = (entityLivingBase.lastTickPosX + (entityLivingBase.posX - entityLivingBase.lastTickPosX) * mc.timer.renderPartialTicks
                - RenderManager.renderPosX)
        val pY = (entityLivingBase.lastTickPosY + (entityLivingBase.posY - entityLivingBase.lastTickPosY) * mc.timer.renderPartialTicks
                - RenderManager.renderPosY)
        val pZ = (entityLivingBase.lastTickPosZ + (entityLivingBase.posZ - entityLivingBase.lastTickPosZ) * mc.timer.renderPartialTicks
                - RenderManager.renderPosZ)
        when (modeValue.get().toLowerCase()) {
            "crystal" -> drawCrystal(killAura.target, pX, pY + 2.25, pZ,if(hurt.get() && killAura.target!!.hurtTime > 3) Color(255, 50, 50, 75).rgb else color.rgb,if(hurt.get() && killAura.target!!.hurtTime > 3) Color(255, 50, 50, 75).rgb else color.rgb)
            "box" -> drawEntityBoxESP(killAura.target,if(hurt.get() && killAura.target!!.hurtTime > 3) Color(255, 50, 50, 75) else color)
            "head" -> drawPlatformESP(killAura.target,if(hurt.get() && killAura.target!!.hurtTime > 3) Color(255, 50, 50, 75) else color)
            "mark" -> drawPlatform(killAura.target,  if(hurt.get() && killAura.target!!.hurtTime > 3) Color(255, 50, 50, 75) else color)
        }
    }

}